src/laser/Laser.H

```diff
@@ -47,14 +172,45 @@ private:
     amrex::Real m_lambda0 {0.}; /**< Laser central wavelength */
     /** Average position of the Gaussian laser pulse */
     amrex::RealVect m_position_mean {0., 0., 0.};
-
+    int m_3d_on_host {0};/** Whether the 3D laser envelope is stored in host or device memory */
+    amrex::Real m_focal_distance {0.};
     /** Number of guard cells for slices MultiFab */
     amrex::IntVect m_slices_nguards = {-1, -1, -1};
+    std::string m_solver_type = "fft";
+    amrex::Box m_slice_box;
 
-    /** Vector over levels, all fields */
-    amrex::Vector<std::array<amrex::FArrayBox, 3>> m_3D_F;
-    /** Vector over levels, array of 4 slices required to compute current slice */
+    /** Nb fields in 3D array: new_real, new_imag, old_real, old_imag */
+    int m_nfields_3d {4};
+    /** 3D laser data. Could be a vector over levels if MR needed for laser */
+    amrex::FArrayBox m_F;
+    /** Array of N slices required to compute current slice */
     std::array<amrex::MultiFab, m_nslices> m_slices;
+    amrex::Real m_MG_tolerance_rel = 1.e-4;
```

Comment:
This should be an input parameter just like in the explicit solver. In single precision and high resolution, 1.e-4 can be too restrictive, so one has to go up to 1e-3 or something like that

Reply:
indeed